cpp/src/io/shp/polygon_shapefile_reader.cu

```diff
@@ -199,14 +199,15 @@ namespace cuspatial
         if (pm.num_feature <=0) return;
 
         cudaStream_t stream{0};
+        auto exec_policy = rmm::exec_policy(stream);    
 
         int32_t* temp{nullptr};
         RMM_TRY( RMM_ALLOC(&temp, pm.num_feature * sizeof(int32_t), stream) );
         CUDA_TRY( cudaMemcpyAsync(temp, pm.feature_length,
                               pm.num_feature * sizeof(int32_t),
                               cudaMemcpyHostToDevice, stream) );
         //prefix-sum: len to pos
-        thrust::inclusive_scan(rmm::exec_policy(stream)->on(stream), temp, temp + pm.num_feature, temp);
+        thrust::inclusive_scan(exec_policy->on(stream), temp, temp + pm.num_feature, temp);
```

Comment:
The old usage of `rmm::exec_policy(stream)->on(stream)` is just fine.

What's problematic is when you do this:

```
auto exec = rmm::exec_policy(stream)->on(stream);
```

Reply:
understood. I wanted to simplify code, especially when exec_policy is used in quite some places.

Reply:
I will change the description to reflex the bug more precisely.